community/modules/scripts/wait-for-startup/scripts/wait-for-startup-status.sh

```diff
@@ -60,14 +60,29 @@ FINISH_LINE="startup-script exit status"
 # Match string for failures on the new guest agent
 FINISH_LINE_ERR="Script.*failed with error:"
 
+NON_FATAL_ERRORS=(
+	"Internal error"
+)
+
 until [[ now -gt deadline ]]; do
 	ser_log=$(
 		set -o pipefail
 		${fetch_cmd} 2>"${error_file}" |
 			c1grep "${FINISH_LINE}\|${FINISH_LINE_ERR}"
 	) || {
-		cat "${error_file}"
-		exit 1
+		err=$(cat "${error_file}")
+		echo "$err"
+		fatal_error="true"
+		for e in "${NON_FATAL_ERRORS[@]}"; do
```

Comment:
Looks like we are adding a risk of running an endless-loop of "NON_FATAL_ERRORS". I guess it's not a big problem since there is a global timeout.

Reply:
Doesn't this just loop through the list of strings?  It's just trying to determine whether the error message contains one of the substrings in $NON_FATAL_ERRORS.

Is there a possibility of an infinite loop that I'm missing elsewhere?